proxy/gateway/nodejs/lib/codec.js

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
```

Comment:
Is there a spec that documents this protocol? Maybe a comment with a link to the spec would be nice.